packages/compass-components/src/components/document-list/element.tsx

```diff
@@ -458,9 +499,39 @@ export const HadronElement: React.FunctionComponent<{
     collapse,
   } = useHadronElement(element);
 
+  // Function to check if a field is in the query
+  // TODO: COMPASS-9541 Improve the functionality when checking for nested objects.
+  const isFieldInQuery = useCallback(
+    (field: string, fieldValue: unknown): boolean => {
+      return hasDistinctValue(
+        query?.[field] as Record<string, unknown>,
+        fieldValue
+      );
+    },
+    [query]
+  );
+
   // Add context menu hook for the field
   const fieldContextMenuRef = useContextMenuItems(
     () => [
+      ...(onUpdateQuery
+        ? [
+            {
+              label: isFieldInQuery(
+                getNestedKeyPathForElement(element),
+                element.generateObject()
+              )
+                ? 'Remove from query'
+                : 'Add to query',
+              onAction: () => {
+                onUpdateQuery(
+                  getNestedKeyPathForElement(element),
+                  element.generateObject()
+                );
+              },
+            },
+          ]
+        : []),
```

Comment:
This component is also used at other places, like in aggregations and how does this work there exactly? I can right click there and I think we should not have this option there, right?

Reply:
yep because `onUpdateQuery` will be undefined then